### ISSUE TITLE
“Direct functions and operators” revision

### ISSUE THREAD

**Issue by RojerGS** [xpqz/learnapl#6]
# Revision

Here I include all of the things that, to the best of my knowledge and experience, should be changed/corrected.

 - [x] "Fortunately, the syntax – if we can even call it that – for user-defined functions, is super-simple." →
   - "Fortunately, the syntax – if we can even call it that –, for user-defined functions, is super-simple." or
   - "Fortunately, the syntax – if we can even call it that – for user-defined functions is super-simple." (IMO this is the best option)
 - [x] "The arrow keys instead moves up and down spatially," → "The arrow keys instead move up and down spatially,"
 - [ ] Maybe I misunderstood the text, but in Dyalog 18.0 you can already do multi-line input, you just have to turn it on explicitly. In RIDE, that's setting the flag `Dyalog_LineEditor_Mode = 1` ([see the docs](http://help.dyalog.com/latest/#UserGuide/Installation%20and%20Configuration/Configuration%20Parameters/Dyalog_LineEditor_Mode.htm?Highlight=dyalog_lineeditor_mode))
 - [x] "This way a function can behave" → "This way, a function can behave"
 - [ ] "On way is" → "One way is"
 - [x] The snippet
```APL
]dinput
foo←{
    a←45
    _←{a←¯99⋄⍬}⍬
    a
}
```
has no spacing around the multiple `←` and the `⋄`, contrary to other snippets.
 - [x] "We could have said:
```APL
      sumred ← +/
      2 sumred ⍳10
1 3 5 7 9 11 13 15 17
```
to emphasize the fact that the operator really returns a function." → 
Drop the ":", or rewrite beginning of next phrase as its own sentence, like "This emphasizes ..."

# To consider

In here I include things that, from my point of view, would be nice additions/modifications. Tick them as you ponder on them, feel free to disregard.

 - [x] Keys can be enclosed in the HTML tag `<kbd>` for an amazing look and feel: <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>Enter</kbd>, for example.
 - [x] Not sure how strict you are being with the glyph names, but `∇` is called "Del", not "Recur".
 - [x] Do a "find+replace" of "boolean" with "Boolean", as the word derives from the proper noun "Boole".
 - [x] When talking about modified assignment, I confess _I personally_ was slightly confused with the final `⋄ ⍬` in `_←{a←¯99⋄⍬}⍬`, `_ ← {a +← 45 ⋄ ⍬}⍬`, and others. Consider removing them (the trailing `⋄ ⍬`).
 - [x]  "is called dyadically, with 2 to its left and `⍳10` to its right." → " is called dyadically, with `2` to its left and `⍳10` to its right." (add code styling to "2" too).

**Comment by xpqz** [comment]
Thank you. Most fixed in c8e63f88305dc9a70304cc8f8444c54f4e06c313 (deployed).

As to the multi-line input in 18.0, as it's marked experimental, requires a config change, and cruciually, is Windows-only (as I understand it), I think I'll rather not mention it.

**Comment by RojerGS** [comment]
Right, I use Windows so I can't vouch for it working on other OSs. It is experimental but the docs don't seem to mention that it's Windows-specific. Either way, you're the boss ○/
